app/models/problems/cancelled_by_vendor.rb

```diff
@@ -0,0 +1,35 @@
+class CancelledByVendor < Problem
+  include ProblemLogic
+
+  before_validation :set_fields
+  validate :validation
+  validates :record_num, uniqueness: true
+  DESCRIPTION = 'Edifact response from vendor indicates cancellation'
+
+  def self.model_name
+    Problem.model_name
+  end
+
+  def validation
+    errors.add(:query, "no match") unless query
+  end
+
+  def query
+    load_order_view
+    order_status_code_is_o and
+    order_received_date_is_nil and
+    order_edi_response_is_2
+  end
+
+  private
+
+  def load_order_view
```

Comment:
This duplicates logic (shared in the r_date problem). We should move this logic into an intermediate class for these to inherit from. E.g.

```ruby
it "should update the SierraIndex.last_checked date" do
        SierraIndexHelper.should receive(:update_date).with(100001, 'o')
        BlankRdate.should receive(:create)
        CancelledByVendor.should receive(:create)
        Resque.enqueue(OrderRecordQc, 100001)
      end
```
class CancelledByVendor < OrderProblem < Problem
class BlankRDate < OrderProblem < Problem